### PR TITLE
SSG pagination example

### PR DESCRIPTION
**Pull request by dominiksipowicz**
This example illustrates how to use SSG/ISR pagination for listing pages to save on build time.

Preview + DEMO: https://pagination-strategy.vercel.app/ 

## Problem:
Long build due to SEO requirement all pages are SSG (`getStaticPaths`)

## Solution:
- On the build phase, only build the first few paginated pages and rest on the fly.
- This example shows how to create SSG Pagination for the first 5 pages, and use ISR for the rest.
- For SEO reasons page 1 is redirected to index page to avoid duplicate content

### CODE REVIEW

File: solutions/pagination-strategy/pages/category/[page].tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    currentPage
+  ]
+
+  if (!currentProducts) {
+    return {
+      notFound: true,
+    }
+  }
+
+  // This can be also set in next.config.js but it's better to keep it here
```
Comment (lfades) [inline solutions/pagination-strategy/pages/category/[page].tsx:59]:
```suggestion
```

File: solutions/pagination-strategy/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "vercel-example--pagination-strategy",
```
Comment (lfades) [inline solutions/pagination-strategy/package.json:2]: Is this an issue with plop? I was expecting the name to be `pagination-strategy`

Comment (dominiksipowicz) [reply solutions/pagination-strategy/package.json:2]: You know. Now I think it's plop issue. When I created plop I thought prefix would be better but I think you're right. it should be just a name without any prefix.

File: solutions/pagination-with-ssg/README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Pagination Strategy example
+# Pagination With SSH example
```
Comment (SuttonJack) [inline solutions/pagination-with-ssg/README.md:1]:
```suggestion
# Pagination With SSG example
```

Comment (dominiksipowicz) [reply solutions/pagination-with-ssg/README.md]: thanks!

File: solutions/pagination-strategy/package.json (hunk 2 above)
Comment (dominiksipowicz) [reply solutions/pagination-strategy/package.json]: https://github.com/vercel/examples/pull/150

**Review by lfades — APPROVED** [review]
Added more changes and it looks good to me!

**Pull request merged**